### PR TITLE
Moar bazel stuff - version and gapic

### PR DESCRIPTION
**Pull request by pmuetschard**
- Carry over defines for swapchain.so from cmake
- Implement the versioning in bazel - we'll edit version.bzl instead of version.cmake
- Add a comment to mm.bzl, pointing out its hackyness
- Fix up the copy actions, since they would corrupt binary files
- Get gapic building
  - GRPC java codegen rule
  - Don't use GeneratedMessage, it's now GeneratedMessageV3 and may soon be something even crazier
  - "Workaround" some error prone warnings
  - Fix a NPE in the crash reporting

I actually managed to run a bazel built gapic, point it at a bazel built gapis & gapir and open a trace. Woot

### CODE REVIEW

File: version.bzl [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            "@GAPID_BUILD_NUMBER@": ctx.var.get("GAPID_BUILD_NUMBER"),
+            "@GAPID_BUILD_SHA@": ctx.var.get("GAPID_BUILD_SHA"),
+        }
+    )
+
+gapid_version = rule(
+    implementation=_gapid_version,
+    attrs = {
+        "template": attr.label(mandatory = True, allow_files = True, single_file = True),
+        "out": attr.output(mandatory = True),
```
Comment (ianthehat) [inline version.bzl]: Implicit outputs are bad and wrong...
Hidden outputs are the only clean answer.
If you want to submit this and then I will show you what I mean in a separate PR I am happy to do that.

File: core/app/BUILD.bazel [hunk 2]
```diff
@@ -29,3 +31,9 @@ go_library(
         "@com_github_pkg_errors//:go_default_library",
     ],
 )
+
+gapid_version(
+    name = "version",
+    out = "default_version.go",
+    template = "default_version.go.in",
```
Comment (ianthehat) [inline core/app/BUILD.bazel]: It would probably be cleaner to do this with x_def stamping.

File: version.bzl (hunk 1 above)
Comment (pmuetschard) [reply version.bzl]: Please do.

File: core/app/BUILD.bazel (hunk 2 above)
Comment (ianthehat) [reply core/app/BUILD.bazel]: The problem with this approach is that it modifies a source at the bottom level, which causes a cascade of rebuilds.
The stamping approach just causes a re-link at the top level.

Comment (pmuetschard) [reply core/app/BUILD.bazel]: I'm OK with that.

**Pull request merged**